### PR TITLE
IRO-1662 - Added TextField and FieldGroup components.

### PR DESCRIPTION
**Pull request by ndemidovich-sc**
Added two new components:
 - `TextField`
 - `FieldGroup`

Provide possibility to customize new components via Chakra UI theme
Added storybook preview for this compoents
TextField

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -28,6 +31,11 @@ module.exports = {
       },
       resolve: {
         ...resolve,
+        alias: {
```
Comment (azhurauski-sc) [inline .storybook/main.js:34]: Look like there is a feature flag that allows fix that problem better way
https://github.com/iron-fish/ui-kit/pull/3/files#diff-cafe2123a72c4ce3a9f7e9ee4b0e188256eb02d7ec1e54fa2cadfac99b22f92bR34

File: components/TextFieldGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        '>:last-child': {
+          borderLeftRadius: '0 !important',
+        },
+      }}
+      alignItems="center"
+    >
+      {Children.toArray(children)
+        .filter(child => isValidElement(child))
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .map((child: any) => {
```
Comment (azhurauski-sc) [inline components/TextFieldGroup.tsx:26]: Can it have type ReactNode?

File: .storybook/preview.js [hunk 3]
```diff
@@ -1,26 +1,26 @@
-import IronFishUIProvider from 'components/IronFishUIProvider';
-import ColorModeSwitcher from 'components/ColorModeSwitcher';
-import { Flex } from "@chakra-ui/react"
+import IronFishUIProvider from 'components/IronFishUIProvider'
+import ColorModeSwitcher from 'components/ColorModeSwitcher'
+import { Box, CSSReset, Flex } from '@chakra-ui/react'
```
Comment (brekk) [inline .storybook/preview.js:3]: is `CSSReset` being used? I'd like to add some pre-commit hooks like `nps lint`, would you and @azhurauski-sc be ok with that?  (I think it'll decrease a bunch of our problems, but I'd be ok with moving it to a github workflow instead -- I personally prefer pre-commit hooks but I don't feel strongly)

File: components/TextField.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }, [value])
+
+  return (
+    <Flex
+      id="TextField"
+      tabIndex={0}
+      onFocus={() => inputRef?.current?.focus()}
+      {...props}
+      sx={{
+        ...styles?.container,
```
Comment (brekk) [inline components/TextField.tsx:43]: Unfamiliar with this offhand -- Am I correct in thinking that the properties here come from [theme.ts](https://github.com/iron-fish/ui-kit/pull/7/files#diff-6ecb222b58bd55a2f5fab1aaccba9193a3f4e92309a433c1e327cce844020834R51) ?

File: components/TextField.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    >
+      {LeftAddons}
+      <Flex flexDirection="column" justifyContent="center" w="100%">
+        <Box sx={styles?.label}>{label}</Box>
+        <Box
+          sx={styles?.inputWrapper}
+          display={!!val ? 'block' : 'none'}
+          _groupFocus={{ display: 'block' }}
+          _groupFocusVisible={{ display: 'block' }}
+          _groupFocusWithin={{ display: 'block' }}
```
Comment (brekk) [inline components/TextField.tsx:58]: Minor nit, but could `{display: 'block'}` be defined as a standalone variable and re-used here and the previous few lines?

File: components/TextField.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        >
+          <Input
+            {...InputProps}
+            ref={inputRef}
+            value={val}
+            variant="unstyled"
+            onChange={e => {
+              setVal(e.target.value)
+              InputProps?.onChange && InputProps?.onChange(e)
```
Comment (brekk) [inline components/TextField.tsx:67]: Nit, but I think no need for the second `?` when invoking?

File: components/TextFieldGroup.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import { Children, cloneElement, FC, isValidElement } from 'react'
+
+const TextFieldGroup: FC<StyleProps> = ({ children, ...props }) => {
+  const styles = useMultiStyleConfig('TextFieldGroup', props)
+  return (
+    <Flex
+      {...props}
+      sx={{
+        ...styles?.container,
+        '>:first-child': {
```
Comment (brekk) [inline components/TextFieldGroup.tsx:11]: 🙌🏽 I wasn't sure how to do this earlier, nice.

File: components/TextFieldGroup.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+          borderLeftRadius: '0 !important',
+        },
+      }}
+      alignItems="center"
+    >
+      {Children.toArray(children)
+        .filter(child => isValidElement(child))
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .map((child: any) => {
+          return cloneElement(child, {
```
Comment (brekk) [inline components/TextFieldGroup.tsx:27]: For my own edification, this is required because of how `@chakra-ui` components / styles are applied?

**Review by brekk — CHANGES_REQUESTED** [review]
Some comments / questions -- can you apply `nps lint`? I think that will help reduce churn, and I'll plan to add it to the github workflow.

**Comment by brekk** [comment]
@ndemidovich-sc Apologies! I was reviewing your branch and then inadvertently pushed a commit (thinking I was in my branch). `39f5fe9` is the commit -- I can force-remove it but since I already made a mistake I didn't want to compound it by forcing more change -- I was imagining you could rebase around it, please LMK if you'd like me to explicitly force-remove it.

File: components/TextField.tsx (hunk 4 above)
Comment (ndemidovich-sc) [reply components/TextField.tsx:43]: Yes, you are correct.

File: .storybook/preview.js (hunk 3 above)
Comment (ndemidovich-sc) [reply .storybook/preview.js]: Good idea about pre-commit hook with `nps lint`.

Comment (azhurauski-sc) [reply .storybook/preview.js]: Also like that idea as I used to rely on IDE lint, but sometimes it might miss something.

File: theme/theme.ts [hunk 9]
```diff
@@ -10,6 +10,94 @@ const typeface: object = {
...
+          h: '4.313rem',
+          bg: mode(NAMED_COLORS.WHITE, NAMED_COLORS.BLACK)(props),
+          borderCollapse: 'collapse',
+          borderRadius: '0.25rem',
+          border: '0.063rem solid',
+          borderColor: mode(
+            NAMED_COLORS.LIGHT_GREY,
+            NAMED_COLORS.DARK_GREY
+          )(props),
+          boxShadow: '0px 4px 11px rgba(0, 0, 0, 0.04)',
```
Comment (azhurauski-sc) [inline theme/theme.ts:26]: Should we use rem here as well?

File: .storybook/main.js [hunk 10]
```diff
@@ -1,4 +1,7 @@
 const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
+const path = require("path");
+
+const toPath = (_path) => path.join(process.cwd(), _path);
```
Comment (azhurauski-sc) [inline .storybook/main.js:4]: This is not needed anymore.

**Review by azhurauski-sc — COMMENTED** [review]
Few minor comments.

**Review by azhurauski-sc — APPROVED** [review]

**Review by brekk — APPROVED** [review]
:shipit: LGTM

**Pull request merged**